### PR TITLE
Expense fields platformv2

### PR DESCRIPTION
**Pull request by Julias0**
Code coverage:

Spender Platform API v1 service:
Screenshot 2023-02-03 at 5 35 03 PM

Expense fields service:
Screenshot 2023-02-06 at 12 55 34 PM

Custom inputs service:
Screenshot 2023-02-06 at 12 55 25 PM

### CODE REVIEW

File: src/app/core/services/expense-fields.service.spec.ts [hunk 1]
```diff
@@ -86,4 +91,12 @@ describe('ExpenseFieldsService', () => {
         done();
       });
   });
+
+  it('should return correct mapping for column name', () => {
+    expect(expenseFieldsService.getColumnName('spent_at')).toBe('txn_dt');
+    expect(expenseFieldsService.getColumnName('locations[0]')).toBe('location1');
+    expect(expenseFieldsService.getColumnName('travel_classes[0]', 2)).toBe('bus_travel_class');
+    expect(expenseFieldsService.getColumnName('travel_classes[1]', 1)).toBe('flight_return_travel_class');
+    expect(expenseFieldsService.getColumnName('random_string')).toBe('random_string');
```
Comment (Julias0) [inline src/app/core/services/expense-fields.service.spec.ts:100]: Why are we handlign the random string case? Shouldnt we throw an error in these cases?

Comment (Julias0) [reply src/app/core/services/expense-fields.service.spec.ts:100]: random_string  -> project
Add a comment that this is for supporting pre-existing normal cases like project, cost center etc

File: src/app/core/services/spender-platform-v1-api.service.spec.ts [hunk 2]
```diff
@@ -1,16 +1,95 @@
...
+
+  it('should make GET request without params', (done) => {
+    httpClient.get.and.returnValue(of(apiResponse));
+
+    spenderPlatformV1ApiService.get('/some_url').subscribe((res) => {
+      expect(res).toEqual(apiResponse);
+    });
+
+    expect(httpClient.get).toHaveBeenCalledWith('https://staging.fyle.tech/platform/v1/spender/some_url', {});
+    done();
```
Comment (Julias0) [inline src/app/core/services/spender-platform-v1-api.service.spec.ts:54]: Move these two lines inside the subscribe block

File: src/app/core/services/spender-platform-v1-api.service.spec.ts [hunk 3]
```diff
@@ -1,16 +1,95 @@
...
+  });
+
+  it('should set root url', () => {
+    expect(spenderPlatformV1ApiService.ROOT_ENDPOINT).toBe(rootUrl);
+  });
+
+  it('should make GET request without params', (done) => {
+    httpClient.get.and.returnValue(of(apiResponse));
+
+    spenderPlatformV1ApiService.get('/some_url').subscribe((res) => {
```
Comment (Julias0) [inline src/app/core/services/spender-platform-v1-api.service.spec.ts:49]: some_url -> replace with expenses/ cost center /something we have

File: src/app/core/services/expense-fields.service.spec.ts (hunk 1 above)
Comment (yksolanki9) [reply src/app/core/services/expense-fields.service.spec.ts:100]: Done

**Review by yksolanki9 — APPROVED** [review]

**Pull request merged**